### ISSUE TITLE
Types are missing in 2.0.x version

### ISSUE THREAD

**Issue by waterplea** [jhildenbiddle/css-vars-ponyfill#69]
Looks like css-vars-ponyfill.d.ts got lost when version was bumped to 2+.

**Comment by waterplea** [comment]
Here's where typings come in handy, btw, with your breaking changes if types were there — we would know immediately what options are gone because an error would be shown. Strong types would also invalidate this:

**watch**
Type: `boolean`
Default: `null`

It would be better to set it explicitly to `false`.

**Comment by jhildenbiddle** [comment]
Thanks for the heads up, @waterplea.

The `.d.ts` file has been [moved](https://github.com/jhildenbiddle/css-vars-ponyfill/blob/master/src/index.d.ts) to the `/src` directory because the `/dist` directory has been removed from the repo in 2.x. My understanding was that so long as I updated the `types` value in `package.json` to point to this file all would be well.

https://github.com/jhildenbiddle/css-vars-ponyfill/blob/d73770c82412b240ba2f7a780c46324cf7046705/package.json#L55

Regarding the default `options.watch` value, `null` is a valid value because this particular option has three possible side effects: `true` creates the MutationObserver, `false` disconnects the MutationObserver, and `null` does nothing. When the MutationObserver calls the ponyfill automatically, it does so with `options.watch` set to `null` so that the existing MutationObserver is neither re-created or disconnected. I believe the docs and `d.ts` file therefore be updated as follows:

```markdown
### watch

- Type: `null|boolean`
- Default: `null`
```

```javascript
watch?: null|boolean;
```

Questions:

1. What is your recommendation on how to restore TypeScript functionality now that the `/dist` directory has been removed? As I mentioned above, the `d.ts` file still exists and the `types` property in `package.json` has been updated, but this setup isn't working.
1. What is your recommendation on how to ensure TypeScript functionality is working before publishing? For example, I created a [demo in StackBlitz](https://stackblitz.com/edit/css-vars-ponyfill-69?file=index.ts) to confirm the issue you've described. Using v1.17.2 of the ponyfill, the editor provides TS information and highlights the error when I set `include:true` (because `include` uses a `string`, not a `boolean`). When I switch to v2.0.1 of the ponyfill, TS information and the error highlight goes away. Makes sense. So, what is the best way for me to make changes and ensure the TS-related issues are fixed before publishing 2.0.2? I know I *could* run `npm pack` on my dev code, create a new project, install the `pack`-ed version of the ponyfill and test, but this is pretty tedious.

Thanks in advance for your assistance.

**Comment by waterplea** [comment]
That is also where strong typings come in handy. You can set loose typings in TypeScript where `null` is part of every type but I promote strict null checks, this way you would know that falsy does not equal false and `null` and `false` do different things. So yes, your suggested edits seem right to me.

As for your questions:
1. You do build the dist somehow with npm commands. You could add a command that would copy `.d.ts` file into the dist folder before you publish it to npm. Some post-build hook or similar idea should be fine. In my libraries I do this for some assets that do not automatically get bundled into dist folder.
2. To really test a distributed version you would need to make a build that goes to npm. You might add a minimal TS setup, a single file that has every parameter passed to cssVars function with correct types. With tsconfig file you can set a folder where it should look for your module as if it was an npm package installed in node_modules. You could run build command, creating a dist folder targeted by TypeScript config file that is identical to what people would get in their node modules and after that run typescript compilation (there is even dry run flag so it will not produce anything, just make sure it compiles). A fact that it compiles would be enough to be sure types are properly set and included. I can try to create this setup tomorrow, I think it shouldn't take much time from me.

**Comment by jhildenbiddle** [comment]
Fixed in 2.0.2.

I am officially awarding myself the facepalm award on this one. Not sure how I missed the fact that placing the `d.ts` file in a folder that isn't published would break TS support. Yeesh. Sorry about that. 🤦‍♂️

As for testing before publishing, I just created a test project with `npm init` and installed my local copy of the ponyfill with `npm i /path/to/ponyfill/`. This allows me to `import` my local copy and verify that hints show up in VSCode as expected. It's not an automated test obviously, but the `d.ts` file doesn't change very often so it's good enough.

Thanks again, @waterplea.
